pkg/devfile/versions/common/types.go

```diff
@@ -0,0 +1,210 @@
+package common
+
+// -------------- Supported devfile project types ------------ //
+// DevfileProjectsType store valid devfile project types
+type DevfileProjectsType string
+
+const (
+	DevfileProjectTypeGit    DevfileProjectsType = "git"
+	DevfileProjectTypeLocal  DevfileProjectsType = "local"
+	DevfileProjectTypeBinary DevfileProjectsType = "binary"
+)
+
+var SupportedDevfileProjectTypes = []DevfileProjectsType{DevfileProjectTypeGit, DevfileProjectTypeLocal, DevfileProjectTypeBinary}
+
+// -------------- Supported devfile component types ------------ //
+// DevfileComponentsType stores valid devfile component types
+type DevfileComponentsType string
```

Comment:
would `DevfileComponentType` be better type name here?

Reply:
Done!